build/util/src/lib.rs

```diff
@@ -215,3 +235,52 @@ fn toml_from_env<T: DeserializeOwned>(var: &str) -> Result<Option<T>> {
         .context("deserializing configuration")?;
     Ok(Some(rval))
 }
+
+pub fn build_notifications() -> Result<()> {
+    let out_dir = out_dir();
+    let dest_path = out_dir.join("notifications.rs");
+    let mut out = std::fs::File::create(dest_path)?;
+
+    let full_task_config = task_full_config_toml()?;
+
+    if full_task_config.notifications.len() >= 32 {
+        bail!(
+            "Too many notifications; \
+             overlapping with `INTERNAL_TIMER_NOTIFICATION`"
+        );
+    }
+    if full_task_config.name == "task-jefe"
+        && full_task_config.notifications.get(0).cloned()
+            != Some("fault".to_string())
+    {
+        bail!("`jefe` must have \"fault\" as its first notification");
+    }
+
+    writeln!(&mut out, "#[allow(dead_code)]")?;
+    writeln!(&mut out, "pub mod notifications {{")?;
+
+    write_task_notifications(&mut out, &full_task_config.notifications)?;
+
+    for task in env_var("HUBRIS_TASKS")
+        .expect("missing HUBRIS_TASKS")
+        .split(",")
+    {
+        let full_task_config = other_task_full_config_toml(task)?;
+        writeln!(&mut out, "pub mod {task} {{")?;
+        write_task_notifications(&mut out, &full_task_config.notifications)?;
```

Comment:
Should we repeat the check for the max number of notifications here? If we add a task with too many and compile _that task_ we'll get the good error message above, but every other task will fail to compile with a less-useful message.

Reply:
Good idea, I added a fallback check in 6b273242a936d5f3cc80102f25d62f00b5c18e23

(this will only matter for _other task IRQs_, so I'm just protecting against `u32` overflow, rather than collision with the default timer IRQ)